src/main/kotlin/app/revanced/patches/ad/VideoAdsPatch.kt

```diff
@@ -0,0 +1,103 @@
+package app.revanced.patches.ad
+
+import app.revanced.patcher.PatcherData
+import app.revanced.patcher.extensions.AccessFlagExtensions.Companion.or
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.*
+import app.revanced.patcher.signature.MethodMetadata
+import app.revanced.patcher.signature.MethodSignature
+import app.revanced.patcher.signature.MethodSignatureMetadata
+import app.revanced.patcher.signature.PatternScanMethod
+import app.revanced.patcher.smali.asInstructions
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
+private val compatiblePackages = arrayOf("com.google.android.youtube")
+
+class VideoAdsPatch : Patch(
+    metadata = PatchMetadata(
+        shortName = "video-ads",
+        name = "YouTube Video Ads Patch",
+        description = "Patch to remove ads in the YouTube video player.",
+        compatiblePackages = compatiblePackages,
+        version = "0.0.1"
+    ),
+    signatures = listOf(
+        MethodSignature(
+            methodSignatureMetadata = MethodSignatureMetadata(
+                name = "show-video-ads-constructor",
+                methodMetadata = MethodMetadata(
+                    definingClass = "zai",
+                    name = "<init>",
+                ),
+                patternScanMethod = PatternScanMethod.Fuzzy(2),// FIXME: Test this threshold and find the best value.
+                compatiblePackages = compatiblePackages,
+                description = """Signature for the constructor of some class.
+                This signature is being used to find another method in the parent class
+                and was discovered in the YouTube version v17.03.38""",
```

Comment:
Make sure to trim the indents (tabs) everywhere you do this.
```suggestion
                and was discovered in the YouTube version v17.03.38""".trimIndent(),
```